calcom/cal.com
# feat: Locations Autocompletion on Event Settings

## What does this PR do?



This PR adds autocomplete from locations, using Google Places API.
Closes #11931 and related



## Requirement/Documentation



This feature requires an api key from [Google Places API](https://console.cloud.google.com/apis/library/places.googleapis.com).

```.env
# .env

# GOOGLE PLACES API
# Needed for Locations Autocomplete in Event Types
GOOGLE_PLACES_API_KEY=
```

## Type of change




## How should this be tested?



- Are there environment variables that should be set?
A Google Places api key needs to be set.
`GOOGLE_PLACES_API_KEY=`
- What are the minimal test data to have?
A string array of places to test the search.
- What is expected (happy path) to have (input and output)?
When user writes a location, a dropdown with relevant options should appear to autocomplete.


https://github.com/calcom/cal.com/assets/45767683/c162f6e4-9c6c-4805-8aa9-f50a64c9399c



## Mandatory Tasks
